helper/ip.c

```diff
@@ -28,3 +28,23 @@ int odph_ipv4_addr_parse(uint32_t *ip_addr, const char *str)
 
 	return 0;
 }
+
+int odph_ipv6_addr_parse(uint8_t * const ip_addr, const char * const str)
+{
+	int i, j;
+	uint16_t hextet[8];
+
+	if ((NULL == ip_addr) || (NULL == str))
+		return -1;
+
+	if (sscanf(str, "%hx:%hx:%hx:%hx:%hx:%hx:%hx:%hx", &hextet[0], &hextet[1], &hextet[2],
+		   &hextet[3], &hextet[4], &hextet[5], &hextet[6],
+		   &hextet[7]) != (ODPH_IPV6ADDR_LEN / 2))
+		return -1;
+
+	for (i = 0, j = 0; i < 16; i += 2, j++) {
+		ip_addr[i] = hextet[j] & 0xFF;
```

Comment:
Byte order looks incorrect. If we consider address "AABB::", 0xBB ends up in the first memory location and 0xAA in the second.

Reply:
Yes.. I will fix this in another change-set